invalid memory address or nil pointer dereference

Prometheus gave the following error after running for a few days:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x77e8e6]

goroutine 184 [running]:
github.com/prometheus/prometheus/retrieval/discovery.(*ConsulDiscovery).watchServices(0xc208451320, 0xc224c14120, 0xc22d0e90e0)
        /app/.build/root/gopath/src/github.com/prometheus/prometheus/retrieval/discovery/consul.go:200 +0x1f6
created by github.com/prometheus/prometheus/retrieval/discovery.(*ConsulDiscovery).Run
        /app/.build/root/gopath/src/github.com/prometheus/prometheus/retrieval/discovery/consul.go:151 +0xbe
```

--- a/retrieval/discovery/consul.go
+++ b/retrieval/discovery/consul.go
@@ -195,6 +195,7 @@ func (cd *ConsulDiscovery) watchServices(update chan<- *consulService, done <-ch
 		if err != nil {
 			log.Errorf("Error refreshing service list: %s", err)
 			time.Sleep(consulRetryInterval)
+			continue
 		}
 		// If the index equals the previous one, the watch timed out with no update.
 		if meta.LastIndex == lastIndex {

